rocAL/source/pipeline/tensor.cpp

```diff
@@ -202,8 +209,49 @@ void Tensor::update_tensor_roi(const std::vector<uint32_t> &width,
             }
         }
     }
+    else if(!_info.is_metadata()) { // Audio - Data
+        auto max_dims = _info.max_shape();
+        unsigned max_samples = max_dims.at(0);
+        unsigned max_channels = max_dims.at(1);
+        auto samples = width;
+        auto channels = height;
+        Roi2DCords *roi = _info.roi().get_2D_roi();
+
+        if (samples.size() != channels.size())
+            THROW("Batch size of Tensor height and width info does not match")
+        if (samples.size() != info().batch_size())
+            THROW("The batch size of actual Tensor height and width different from Tensor batch size " + TOSTR(samples.size()) + " != " + TOSTR(info().batch_size()))
+        for (unsigned i = 0; i < info().batch_size(); i++) {
+            if (samples[i] > max_samples) {
+                ERR("Given ROI width is larger than buffer width for tensor[" + TOSTR(i) + "] " + TOSTR(samples[i]) + " > " + TOSTR(max_samples))
+                roi[i].xywh.x = max_samples;
+            }
+            else {
+                roi[i].xywh.x = samples[i];
+            }
+            if (channels[i] > max_channels) {
+                ERR("Given ROI height is larger than buffer with for tensor[" + TOSTR(i) + "] " + TOSTR(channels[i]) + " > " + TOSTR(max_channels))
+                roi[i].xywh.y = max_channels;
+            }
+            else {
+                roi[i].xywh.y = channels[i];
+            }
+        }
+    }
+}
+
+void Tensor::update_audio_tensor_sample_rate(const std::vector<float> &sample_rate) {
+    if (_info.is_image()) {
+        THROW("No sample rate available for Image data")
+    }
+    else if(!_info.is_metadata()) {
+        for (unsigned i = 0; i < info().batch_size(); i++) {
+            _info.get_sample_rate()->at(i) = sample_rate[i];
```

Comment:
Can we use a direct memcpy here instead of manually assigning element by element